src/Data/Nat/Properties.agda

```diff
@@ -1308,6 +1299,66 @@ m≤n⇒m∸n≡0 : ∀ {m n} → m ≤ n → m ∸ n ≡ 0
 m≤n⇒m∸n≡0 {n = n} z≤n      = 0∸n≡0 n
 m≤n⇒m∸n≡0 {_}    (s≤s m≤n) = m≤n⇒m∸n≡0 m≤n
 
+m∸n≢0⇒n<m :  {m n : ℕ} → m ∸ n ≢ 0 → n < m
+m∸n≢0⇒n<m {m} {n} m∸n≢0  with n <? m
+... | yes n<m =  n<m
+... | no n≮m =  contradiction m∸n≡0 m∸n≢0
+               where
+               m≤n = ≮⇒≥ n≮m;   m∸n≡0 = m≤n⇒m∸n≡0 m≤n
+
```

Comment:
These should be inlined IMO.

And if you're going to cast `¬ (m < n)` to `n ≥ m` immediately after getting the result of
the decision procedure, it may be worth introducing the intermediate lemma:

```agda
<-≥-connex : Conn _<_ _≥_
```

where [this is `Conn`](https://agda.github.io/agda-stdlib/Relation.Binary.Core.html#3514) and use that instead.

Reply:
> These should be inlined IMO.

I "grepped" through the master branch of April 12 and have not found any ``inline`` pragma there. 
In Glasgow Haskell system, I understood what does it mean ``inline``. And I did not use it there explicitly, because the ``-O`` option inlines small things automatically. 
And I do not understand the meaning of ``inline`` in Agda. I suspected that it inlines everything.
(?)

> it may be worth introducing the intermediate lemma:
```
     <-≥-connex : Conn _<_ _≥_
```
We have the function ``≮⇒≥`` in standard. It is applied in a simplest way, and produces of what it is needed. Why do we need to add a proof for ``Conn _<_ _≥_`` ?
I expect, the things need to be done in a possibly simple way.

Reply:
By inline I mean manually inlining, not adding a pragma.

> Kind of an unneded complication.

IMO the unneeded complication is to use the wrong principle and then
immediately cast the result to what you *actually* wanted.